velox/functions/lib/window/NthValue.cpp

```diff
@@ -126,35 +138,54 @@ class NthValueFunction : public exec::WindowFunction {
     }
 
     if (constantOffset_.has_value()) {
-      setRowNumbersForConstantOffset(
-          ignoreNullsForBlock,
-          validRows,
-          rawFrameStarts,
-          rawFrameEnds,
-          leastFrame);
+      if (std::holds_alternative<int32_t>(constantOffset_.value())) {
+        setRowNumbersForConstantOffset<int32_t>(
+            ignoreNullsForBlock,
+            validRows,
+            rawFrameStarts,
+            rawFrameEnds,
+            leastFrame);
+      } else {
+        setRowNumbersForConstantOffset<int64_t>(
+            ignoreNullsForBlock,
+            validRows,
+            rawFrameStarts,
+            rawFrameEnds,
+            leastFrame);
+      }
     } else {
-      setRowNumbersForColumnOffset(
-          ignoreNullsForBlock,
-          numRows,
-          validRows,
-          rawFrameStarts,
-          rawFrameEnds,
-          leastFrame);
+      if (std::holds_alternative<FlatVectorPtr<int32_t>>(offsets_.value())) {
```

Comment:
Do we need to use std::holds_alternative ? We can get the type of the vector from it directly.

Reply:
Reverted use of variants